### PR TITLE
Reshape vehicle occupancies

### PR DESCRIPTION
**Pull request by julyto**
In the same way of an other ap, I isolated the vehicle occupancies code.

```
forseti
|
|---cmd
|     |---main.go
|---internal
|     |---vehiculeoccupancies
|     |    |---context.go
|     |    |---api.go
|     |    |---vehiculeoccupancies.go
|     |    |---vehiculeoccupancies_test.go
|     |    |---prometheus.go
|     | ...
|     | ...
|---Dockerfile
|---Makefile
|...
```

I also moved api.go in an API module and the manager of all the api's joined them in Internal module.

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -70,11 +72,17 @@ func StatusHandler(manager *DataManager) gin.HandlerFunc {
...
+		var loadVehicleOccupanciesData bool = false
+		if manager.GetVehiculeOccupanciesContext() != nil {
+			// manage vehicleoccupancy activation /status?vehicle_occupancies=true or false
+			vehicleOccupancyStatus := c.Query("vehicle_occupancies")
+			if len(vehicleOccupancyStatus) > 0 {
+				toActive, _ := strconv.ParseBool(vehicleOccupancyStatus)
+				manager.GetVehiculeOccupanciesContext().ManageVehicleOccupancyStatus(toActive)
+			}
+			lastVehicleOccupanciesDataUpdate = manager.GetVehiculeOccupanciesContext().GetLastVehicleOccupanciesDataUpdate()
+			loadVehicleOccupanciesData = manager.GetVehiculeOccupanciesContext().LoadOccupancyData()
```
Comment (benoit-bst) [inline api/api.go:85]: @kadhikari We need your expertise here :mag:

File: api/api_test.go [hunk 2]
```diff
@@ -1,8 +1,9 @@
-package forseti
+package api
```
Comment (benoit-bst) [inline api/api_test.go:1]: Great idea to change the package name for that !
it's more precise

File: internal/manager/manager.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
+package manager
```
Comment (benoit-bst) [inline internal/manager/manager.go:1]: :+1:

**Review by benoit-bst — APPROVED** [review]
This was a huge work for a first PR :clap: 
I think it is a good idea to reshape like that, for more readability and avoid the monolithic effect 

You dove into the code like a jedi !


_Just a question_:

Have you run the magical line to start all services with real data ?

like that `./forseti --departures-uri file:///path/to/extract_edylic.txt --departures-refresh=1s --parkings-uri file:///forseti/fixtures/parkings.txt --parkings-refresh=30s --equipments-uri file:///forseti/fixtures/NET_ACCESS.XML --equipments-refresh=60s --occupancy-files-uri file:////NAVP-1750_vehicle_occupancies/data --occupancy-refresh=300s --occupancy-navitia-uri https://api.navitia.io/v1/coverage/fr-idf --occupancy-navitia-token token --occupancy-service-uri https://geteventsbyclientapi-rec-func-apim.azure-api.net --occupancy-service-token token --occupancy-service-refresh-active true --free-floatings-uri "https://flow-api.fluctuo.com" --free-floatings-token token --free-floatings-refresh=100s --free-floatings-refresh-active true`

**Comment by benoit-bst** [comment]
:bulb: On this PR or inside an other:

Can you create a readme like https://github.com/CanalTP/forseti/blob/master/internal/equipments/readme.md inside the folder to explain the services/Api ?

With a link into the main doc :

**Comment by julyto** [comment]
> This was a huge work for a first PR 
> I think it is a good idea to reshape like that, for more readability and avoid the monolithic effect
> 
> You dove into the code like a jedi !
> 
> 
> _Just a question_:
> 
> Have you run the magical line to start all services with real data ?
> 
> like that `./forseti --departures-uri file:///path/to/extract_edylic.txt --departures-refresh=1s --parkings-uri file:///forseti/fixtures/parkings.txt --parkings-refresh=30s --equipments-uri file:///forseti/fixtures/NET_ACCESS.XML --equipments-refresh=60s --occupancy-files-uri file:////NAVP-1750_vehicle_occupancies/data --occupancy-refresh=300s --occupancy-navitia-uri https://api.navitia.io/v1/coverage/fr-idf --occupancy-navitia-token token --occupancy-service-uri https://geteventsbyclientapi-rec-func-apim.azure-api.net --occupancy-service-token token --occupancy-service-refresh-active true --free-floatings-uri "https://flow-api.fluctuo.com" --free-floatings-token token --free-floatings-refresh=100s --free-floatings-refresh-active true`

**Comment by julyto** [comment]
Yes @benoit-bst I used the magical line to run all services with real life !
I will add a readme in api package and put the link into the main doc.

**Review by benoit-bst — APPROVED** [review]

**Review by kadhikari — CHANGES_REQUESTED** [review]
Can't we move code for data.go into context.go for each api?

Great work :+1:

**Comment by benoit-bst** [comment]
@kadhikari, I think the data inside data.go is shared between several API. This is reason why I created this file. 
To check after that work

**Review by kadhikari — APPROVED** [review]

**Pull request merged**